File: readability.py
def main():

    # get user text input
    text = input("Text: ")

    # get the counts of letters, words, and sentences
    letters = letter_count(text)
    words = word_count(text)
    sentences = sentence_count(text)

    # Calculating the grade reading level
    gradeLevel = round((0.0588 * (letters / words) * 100) -
                       (0.296 * (sentences / words) * 100) - 15.8)

    # Printing the grade level
    if gradeLevel < 1:
        print("Before Grade 1")
    elif gradeLevel >= 1 and gradeLevel < 16:
        print("Grade", gradeLevel)
    elif gradeLevel >= 16:
        print("Grade 16+")


def letter_count(s):
    l = 0
    for i in range(len(s)):
        if s[i].isalpha():
            l += 1
    return l


def word_count(s):
    w = 0
    for i in range(len(s)):
        if s[i].isspace():
            w += 1
    wor = w + 1
    return wor


def sentence_count(s):
    sen = 0
    for i in range(len(s)):
        if s[i] in ["?", ".", "!"]:
            sen += 1
    return sen


main()

File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int letter_count(string s);
int word_count(string s);
int sentence_count(string s);

int main(void)
{
    // promt the user for text
    string tex = get_string("Text: ");
    // create a function to get the grade level
    int let = letter_count(tex);
    int wor = word_count(tex);
    int sen = sentence_count(tex);

    float gl = (0.0588 * ((float) let / (float) wor) * 100) -
               (0.296 * ((float) sen / (float) wor) * 100) - 15.8;
    gl = round(gl);
    int gradelevel = gl;
    // print the grade level
    if (gl < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (gl >= 1 && gl < 16)
    {
        printf("Grade %i\n", gradelevel);
    }
    else if (gl >= 16)
    {
        printf("Grade 16+\n");
    }
}

int letter_count(string s)
{
    int l = 0;
    int i;
    int n = 0;

    for (i = 0, n = strlen(s); i < n; i++)
    {
        if (isalpha(s[i]) != 0)
        {
            l++;
        }
    }
    return l;
}

int word_count(string s)
{
    int w = 0;
    int i;
    int n = 0;

    for (i = 0, n = strlen(s); i < n; i++)
    {
        if (isspace(s[i]) != 0)
        {
            w++;
        }
    }
    int wor = w + 1;
    return wor;
}

int sentence_count(string s)
{
    int sen = 0;
    int i;
    int n = 0;

    for (i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == '?' || s[i] == '.' || s[i] == '!')
        {
            sen++;
        }
    }
    return sen;
}
